### ISSUE TITLE
`generation_logits` gives different values when top_p is on

### ISSUE THREAD

**Issue by dontloo** [NVIDIA/TensorRT-LLM#879]
I'm using v0.6.1 version, the `generation_logits` in the [`outputs`](https://github.com/NVIDIA/TensorRT-LLM/blob/v0.6.1/examples/gpt/run.py#L324) of `generate` method gives different values when `top_p` is on, my observation is that with top_p enabled it stores probs instead of logits, is that by design?

**Comment by byshiue** [comment]
Could you share the results of different cases? What's difference do you observe?

**Comment by dontloo** [comment]
generation_logits  on the sample generation
```
top_p on [0.9969, 1.0000, 1.0000, 1.0000, 1.0000, 1.0000, 1.0000]
top_p off temperature=1 [209.0000, 302.5000, 307.2500, 289.5000, 369.7500, 299.5000, 326.0000]
top_p off temperature=16 [13.0625, 18.9062, 19.2031, 18.0937, 23.1094, 18.7187, 20.3750]
```

**Comment by dontloo** [comment]
my speculation is that the buffer that stores logits values is reused by sampling kernels, so it gives different output for different sampler configs

**Comment by byshiue** [comment]
Could you try adding `--use_py_session` when you run the example?

**Comment by nv-guomingz** [comment]
hi do u still have further issue or question now? If not, we'll close it soon.
